src/main/java/com/minecolonies/util/SchematicWrapper.java

```diff
@@ -402,12 +402,13 @@ private static Schematic scanSchematic(World world, BlockPos from, BlockPos to)
      *
      * @param schematic Schematic to add the TileEntity to.
      * @param tileEntity The tile entity.
-     * @param newPos The schematic pos of the tile entity.
+     * @param minPos The schematic min pos to subtract of the tile entity.
      */
-    private static void saveTileEntity(Schematic schematic, TileEntity tileEntity, BlockPos newPos)
+    private static void saveTileEntity(Schematic schematic, TileEntity tileEntity, BlockPos minPos)
     {
         if (tileEntity != null)
         {
+            BlockPos newPos = tileEntity.getPos().subtract(minPos);
             NBTTagCompound tileEntityNBT = new NBTTagCompound();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'tileEntityNBT' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)